modules/sievefilters/functions.php

```diff
@@ -285,13 +285,13 @@ function get_sieve_client_factory($site_config)
 }
 
 if (!hm_exists('prepare_sieve_script ')) {
-    function prepare_sieve_script ($script, $index = 1, $action = "decode")
+    function prepare_sieve_script ($script, $index = 1, $action = "decode", $object_to_array = true)
```

Comment:
It is better to standardize this - either always return objects or arrays when decoding. Also checking calling code that it works fine with whatever type you choose to use here.